include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -8793,4 +8793,32 @@ def err_typecheck_non_count_return_bounds : Error<
 def err_typecheck_bounds_type_annotation_identifier: Error<
   "type name cannot have identifier in it">;
 
+def err_typecheck_bounds_type_annotation_for_illegal_type : Error<
+  "interface type only allowed for a declaration with pointer "
```

Comment:
When we refer to "interface type" in the errors, do you think we should further qualify and instead indicate "Checked C interface type"?  Or something of that vein?  I'm considering the case of someone who tweaks code in an existing Checked C code base, isn't terribly familiar with Checked C, and sees this error -- wondering what an "interface type" is in C.